reframe/frontend/executors/__init__.py

```diff
@@ -197,6 +202,25 @@ def _environ_supported(self, check, environ):
         else:
             return ret and check.supports_environ(environ.name)
 
+    def _retry_failed(self, checks, system):
+        while (self._stats.num_failures() and
+               self._current_run < self._max_retries):
+            failed_checks = [
+                c for c in checks if c.name in
+                set([t.check.name for t in self._stats.tasks_failed()])
+            ]
+            self._current_run += 1
+            self._stats.next_run()
+            if self._stats.current_run != self._current_run:
+                    raise AssertionError('current_run variable out of sync')
```

Comment:
You could print the two values here to help debugging.

Reply:
Done.